### ISSUE TITLE
[Bug]: Changing Fonts or Colors in Customizer Breaks Fonts dropdown in Global Styles in FSE Themes

### ISSUE THREAD

**Issue by druesome** [Automattic/wp-calypso#65115]
### Quick summary

Changing fonts or colors in the Customizer results in only two fonts appearing in the fonts dropdown in Global Styles in FSE Themes.

<img src="https://d.pr/i/VjVQb3+" width="700" />

Reported in 35926348-hc
Follow-up Ticket: [5336007](https://href.li/?5336007-zd-woothemes)

### Steps to reproduce

1. Activate an FSE theme (Meraki or Antonia) on a test site (Simple or Atomic).
2. Hover mouse over Appearance and open Additional CSS in new tab.
3. Click Appearance > Site Editor.
4. Click Global Styles icon on the top right.
5. Click Typography > Text > Font Family and verify that the dropdown displays a full list of fonts.
6. Switch to tab with the Customizer open.
7. Click back and click Fonts or Colors.
8. Select any font under Body font and Heading font under Fonts, or change any color (Primary, Secondary, etc) under Colors.
9. Save Changes.
10. Go back to Site Editor and refresh page.
11. Check fonts dropdown in Global Styles > Typography > Text > Font Family.

### What you expected to happen

I expect the full list of fonts to appear:

<img src="https://d.pr/i/E8TNB1+" width="700" />

### What actually happened

Only the two fonts I selected in Customizer > Fonts appear:

<img src="https://d.pr/i/VjVQb3+" width="700" />

This happens on both Atomic and Simple sites using FSE themes.  I have tested this on non-FSE themes with Global Styles but couldn't reproduce as it seems that the Fonts and Colors sections in the Customizer are restricted.

### Browser

Google Chrome/Chromium, Mozilla Firefox

### Simple/Atomic

Simple, Atomic

### Reproducibility

Consistent

### Available workarounds?

Yes, difficult to implement

### Workaround details

On Atomic sites, it is possible to fix the issue by going to the database and editing the contents of the `wp_global_styles` post of the active theme to `{"version": 2, "isGlobalStylesUserThemeJSON": true }` and then flushing the cache in CLI.  This workaround isn't possible on Simple sites.

**Comment by mriyazuddin** [comment]
Similar issue reported here: 5339315-zd-woothemes. I followed the workaround and that fixed the issue.

**Comment by Joshgrygiel** [comment]
📌 HOUSEKEEPING
- Labels ✅ 
- Priority ✅ 
- Replicable on Core - No

📌 SCRUBBING
- Tested on P2 - N/A
- Tested on Simple ✅ 
- Tested on AT ✅ 
- Tested on Self-hosted - N/A

📌 FINDINGS/SCREENSHOTS/VIDEO
- I can confirm the fonts selection after choosing a Body/Heading font, or changing the font colors through the Customizer does get reduced to only two selections within the Global Styles typography settings

![Markup 2022-07-05 at 11 35 31](https://user-images.githubusercontent.com/88345647/177375288-2b969658-1c7c-4e61-99a1-4e1201b7ae59.png)

- Tested on Meraki and Antonia and could replicate the issue on both themes, on both Simple and AT
- The workaround details did resolve the font selection issue on the AT site for both themes as well

📌 ACTIONS
- Marked as Triaged for Quality Squad review

**Comment by nagpai** [comment]
Maybe related - 36044582-hc
Changing tertiary color via customizer does nothing 
Theme: Payton

**Comment by kerrynicl** [comment]
Also reported here - -5413976-zen . I disabled the gutenberg plugin which seemed to help

**Comment by cometgrrl** [comment]
@Automattic/team-calypso Seems like this might be Gutenberg related, can y'all take a look?

**Comment by chad1008** [comment]
I've been looking at this this morning and I'm finding it hard to reliably reproduce. In my tests FSE themes like Meraki or Antonia don't actually _let_ me modify fonts in the Customizer... that panel just has a notice directing me back to the site editor to manage fonts.

I was able to reproduce it on my Sandbox, though I'm honestly not sure at this point what combination of theme/font changes triggered it. On my non-sandboxed site though, I can't reproduce so far.

**Comment by rw-ye** [comment]
Another report here: 5426707-zd-woothemes


Theme: Videomaker White

I noticed that immediately after switching to the Videomaker White theme, none of the font options show in the dropdown for Global Styles and under the Headings block. 

I found that switching to Videomaker brought all the font options back on both Simple and AT sites.

**Comment by Robertght** [comment]
@rw-ye The one above with Videomaker White is related to this bug: https://github.com/Automattic/wp-calypso/issues/66008 

For now, related to this one, I'm cc'ing @pbking for a second look.

**Comment by fstath** [comment]
Another report: https://wordpress.com/forums/topic/font-disapperance/?view=all
Theme: Meraki

**Comment by cuemarie** [comment]
I've encountered this again here: 36456158-hc
- [ ] 5476228-zd-woothemes
- Theme: **Meraki**

In this user's case, there are different Fonts shown that the examples above: 

<img width="400" alt="screenshot" src="https://user-images.githubusercontent.com/27249804/184504864-49faf1ab-50f8-4ad8-b1aa-20ba9c6a2bce.png">

I followed the steps to solve above (for both **Meraki** and **Russell**, an inactive theme that was also showing the issue), however the fonts still only showed 2 options in the `Editor > Styles > Typography` section. This might be a different issue, but wanted to note here.

**Comment by essleeung** [comment]
Another case here: 7091938-zd-a8c and used the workaround steps: p9F6qB-9Ic-p2#comment-47863

**Comment by annezazu** [comment]
I can't replicate this using a block theme (onyxpulse) and WP 6.6.1 on a simple site with Gutenberg installed:

https://github.com/user-attachments/assets/fd453abb-4de2-49e1-b295-ef9100d52226

To be safe, I tried testing with Meraki and I still cannot replicate. Note that in the customizer fonts and colors push me to the site editor now:


https://github.com/user-attachments/assets/431ea3ed-87c2-4266-ba1a-83b3d85fc8f2


Closing this out as a result. Happy to reopen if folks have further reports or can replicate.
